arangod/Aql/ExecutionNode.cpp

```diff
@@ -2824,14 +2824,14 @@ ExecutionNode* AsyncNode::clone(ExecutionPlan* plan, bool withDependencies,
 }
 
 namespace {
-const char* MATERIALIZE_NODE_IN_NM_COL_PARAM = "inNmColPtr";
-const char* MATERIALIZE_NODE_IN_NM_DOC_PARAM = "inNmDocId";
-const char* MATERIALIZE_NODE_OUT_VARIABLE_PARAM = "outVariable";
+constexpr std::string_view MATERIALIZE_NODE_IN_NM_DOC_PARAM = "inNmDocId";
+constexpr std::string_view MATERIALIZE_NODE_OUT_VARIABLE_PARAM = "outVariable";
+constexpr std::string_view MATERIALIZE_NODE_MULTI_NODE_PARAM = "multiNode";
 }  // namespace
 
 MaterializeNode* materialize::createMaterializeNode(
-    ExecutionPlan* plan, arangodb::velocypack::Slice const& base) {
-  if (base.hasKey(MATERIALIZE_NODE_IN_NM_COL_PARAM)) {
+    ExecutionPlan* plan, arangodb::velocypack::Slice const base) {
+  if (base.hasKey(MATERIALIZE_NODE_MULTI_NODE_PARAM)) {
```

Comment:
Nit: here we only check if the key is present, but we don't check its value.
It is at least theoretically possible that in the future someone changes the code so we will send `{ "multiNode": false }`, and will wrongly interpret it and creating a multi node.

Reply:
True. Fixed